pyp2rpm/metadata_extractors.py

```diff
@@ -219,167 +332,131 @@ def data_from_archive(self):
+                                        else settings.DEFAULT_PYTHON_VERSION
+        archive_data['python_versions'] = py_vers[1:] if py_vers \
+                                            else [settings.DEFAULT_ADDITIONAL_VERSION]
+
+        archive_data['doc_files'] = self.doc_files
+        archive_data['py_modules'] = self.py_modules
+        archive_data['has_test_suite'] = self.has_test_suite
+        archive_data['has_bundled_egg_info'] = self.has_bundled_egg_info
+        archive_data['has_packages'] = self.has_packages
+        archive_data['packages'] = self.packages
+        
         sphinx_dir = self.sphinx_dir
         if sphinx_dir:
             archive_data['sphinx_dir'] = "/".join(sphinx_dir.split("/")[1:])
             archive_data['build_deps'].append(
                 ['BuildRequires', 'python-sphinx'])
 
-        archive_data['has_packages'] = self.has_packages
-        archive_data['py_modules'] = self.py_modules
-        archive_data['scripts'] = self.scripts
-        archive_data['license'] = self.license_from_archive
-        archive_data['packages'] = self.packages
-        archive_data['has_test_suite'] = self.has_test_suite
-        py_vers = self.versions_from_archive
-        archive_data['base_python_version'] = py_vers[0] if py_vers \
-                                        else settings.DEFAULT_PYTHON_VERSION
-        archive_data['python_versions'] = py_vers[1:] if py_vers \
-                                        else [settings.DEFAULT_ADDITIONAL_VERSION]
-
         return archive_data
 
-    @property
-    def data_from_venv(self):
-        """Returns all metadata extractable from virtualenv object.
-        Returns:
-            dictionary containing metadata extracted from virtualenv
-        """
-        if not self.venv:
-            return {}
 
+class DistMetadataExtractor(SetupPyMetadataExtractor):
+    """Metadata extractor based on bdist_rpm distutils command"""
+
+    def __init__(self, *args, **kwargs):
+        super(DistMetadataExtractor, self).__init__(*args, **kwargs)
+       
         temp_dir = tempfile.mkdtemp()
         try:
-            extractor = virtualenv.VirtualEnv(self.name, temp_dir,
-                                              self.name_convertor, 
-                                              self.base_python_version)
-            return extractor.get_venv_data
-        except VirtualenvFailException:
-             logger.error("Skipping virtualenv metadata extraction")
-             return {}
+            with self.archive as a:
+                a.extract_all(directory=temp_dir)
+                try:
+                    setup_py = glob.glob(temp_dir + "/{0}*/".format(self.name) + 'setup.py')[0]
+                except IndexError:
+                    print("setup.py not found; maybe local_file is not proper source archive")
+                    logger.error("setup.py not found, metadata extraction failed.")
```

Comment:
Exiting the tool with error code and without any message to user doesn't seem to be correct behaviour (maybe it should be printed to stderr instead). And log would be incomplete without information like this.What solution do you propose?